src/weppy.ts

```diff
@@ -0,0 +1,478 @@
+/// <reference path="../weppy.d.ts"/>
+
+// these seem not to be defined in window
+interface Window {
+	XMLHttpRequest?: XMLHttpRequest;
+	XDomainRequest?: XMLHttpRequest;
+	JSON?: JSON;
+}
+
+module WeppyImpl {
+
+	var protocolVersion = 3,
+		pathDelimiter = '.',
+		namespaceDelimiter = '::',
+		active = false,
+		options: WeppySettings = {
+			"host": '/weppy',
+			"transport": 'url', // 'url' or 'post'
+			"active": true,
+			"sample": 0.01,
+			"aggregationInterval": 1000,
+			"maxInterval": 5000,
+			"decimalPrecision": 3,
+			"page": 'index',
+			"context": {},
+			"debug": false
+		},
+		initTime = +(new Date),
+		queue, aggregationTimeout, maxTimeout, sentPerformanceData,
+		now = () => {
+			return window.performance && window.performance.now ? window.performance.now() : +(new Date);
+		},
+		log: any = () => {
+			if (options.debug) {
+				if (typeof options.debug == 'function') {
+					options.debug.apply(window, arguments);
+				} else {
+					window.console && window.console.log && window.console.log.apply
```

Comment:
We don't support any browsers that don't support console.log, additionally, console.log fallbacks exist in many other libraries and codebases, so no need to check for it here.